src/foam/u2/view/ScrollDAOView.js

```diff
@@ -341,9 +245,273 @@ foam.CLASS({
     }
   ],
 
+  classes: [
+    {
+      name: 'Row',
+      extends: 'foam.u2.Element',
+
+      documentation: `Recycled DOM rows. The "view" ViewSpec is used to manually
+          tear down and rebuild row contents whenever "data" changes. Scroll
+          performance is optimized by using an inner view with no dynamic
+          bindings so that the only DOM operation is the row completely
+          replacing its contents.`,
+
+      imports: [ 'rowFormatter' ],
+
+      axioms: [
+        foam.u2.CSS.create({
+          code: function CSS() {/*
+            ^ {
+              display: block;
+              contain: layout;
+              will-change: transform;
+              padding: 5px;
+              box-sizing: border-box;
+            }
+            ^ * {
+              background-color: #dddddd;
+            }
+        */}
+        })
+      ],
+
+      properties: [
+        [ 'nodeName', 'li' ],
+        {
+          name: 'data',
+          postSet: function(old, nu) {
+            if ( this.state !== this.LOADED ) return;
+            this.el().innerHTML = this.rowFormatter.format(nu);
```

Comment:
Re: formatter using U2. That was actually what I implemented first, but it was too slow. I'm also quite unhappy that this is what I had to resort to for good performance (with U2 as-is today).

I would be thrilled to see the U2 output cycle (for U2 formatters) or slot bindings (for "plain U2 views") optimized enough to get rid of this. I think it's a useful view to expose through the framework. Unfortunately, I don't have the resources to spend on U2 optimization right now. (Not that I'm the best person for the job anyway; I was never "the U2 expert".)